examples/SerialIface/SerialIface.ino

```diff
@@ -24,6 +24,12 @@
  * PC:      0xb80a014f02 (write OPL register and delay)
  * Arduino: k
  *
+ * A variant of this protocol is available without the delays. In this
+ * case, the BUF? command should be sent as B0F? The binary protocol
+ * is now using 2-byte binary commands:
+ *   - (uint8)  OPL2 register address
+ *   - (uint8)  OPL2 register data
```

Comment:
IMO this should be the only mode of the serial interface.

Reply:
Just to give some context, this was the first protocol I tried when I first wrote the player. I couldn't achieve smooth playback in some cases using Python and pyserial though.

That said, @vincentbernat was able to get great results with the two-byte protocol and a DOSBox patch, so it is possible. May require rewriting the player in C though, which would mean dealing with platform differences around the serial port (or using a library which deals with that).

Reply:
Sure and it made perfect sense when just playing DRO, IMF and VGM files :). But as it's used in different scenarios I think it's better to update the protocol to be more flexible rather than adding on a variant for this case and maybe in the future another one. No reason why Python shouldn't be able to achieve smooth playback. I can have a look at that in the coming days, hough I'm not a Python expert...

Reply:
Thanks - if you can find a way to make it work I definitely agree that one simple protocol is better than two.

Reply:
I have code to this a separate thread to receive acks. In my tests, it didn't make a difference. It's here: https://github.com/vincentbernat/ArduinoOPL2/tree/feature/ack-thread.

IMO, it's sensible to have the two ways: one way where you can buffer data because you know everything in advance and you get more precise timings and another where you need to stream data in real time. If it makes the example more complex for the beginners, we may split the example in two.